### ISSUE TITLE
Triggers in a multi-schema DB setup are pruned incorrectly

### ISSUE THREAD

**Issue by ooprala** [Opus10/django-pgtrigger#54]
Hi, this is going to be a longer description, so please bear with me :)

TL;DR: The [query](https://github.com/Opus10/django-pgtrigger/blob/3ef7e0361901d0b9548ad526768a2fe5412e5d1d/pgtrigger/core.py#L916) which pgtrigger uses to find all its installed triggers unfortunately responds with fully qualified table names for all other schemas (that are not in the search path).

Let's assume my schemas/tenants within a single database are called `foo` and `bar` and that I have a Django app `app` and a model `Customer`, with an associated migration and table run for each of these tenants. Lastly, let's have a trigger that runs some logic on row insertion.

Then, as part of pgtrigger's post-migrate action,  existing triggers will eventually be queried and checked for possible pruning.
However, for something like:
```sql
SET SEARCH_PATH TO foo;
SELECT tgrelid::regclass, tgname, tgenabled FROM pg_trigger WHERE tgname LIKE 'pgtrigger_%';
```
...the result from Postgres would be along these lines:

| tgrelid | tgname | tgenabled |
| :--- | :--- | :--- |
| app\_customer | pgtrigger\_name | O |
| bar.app\_customer | pgtrigger\_name | O |

A few lines later, `_drop_trigger` will try to [fetch a corresponding model](https://github.com/Opus10/django-pgtrigger/blob/3ef7e0361901d0b9548ad526768a2fe5412e5d1d/pgtrigger/core.py#L65) for the DB table `bar.api_customer`.
However, Django itself is blissfully ignorant of DB schemas, ultimately returning `None` instead of the proper model.

This results in pgtrigger dropping the triggers in all but the current schema. In a multi-tenant environment like the one I've described, only the last tenant to be migrated would be the only one whose triggers would remain unpruned.

I realise this is quite a handful. I'll gladly provide a more technical example or a PR, if that would be preferable. Thank you!

**Comment by wesleykendall** [comment]
@ooprala thanks for the detailed description of this. Makes total sense. This seems to be the behavior that also is the cause of https://github.com/Opus10/django-pgtrigger/issues/44.

I unfortunately didn't consider schemas when designing the library, but it sounds like I may be able to fix this issue pretty easily. Will report back here after I try to reproduce

**Comment by wesleykendall** [comment]
@ooprala do you use a particular Django app to manage schemas? If I'm understanding correctly, something sets the search path so that Django always operates under the correct schema. Or does it use a DB router? https://github.com/Opus10/django-pgtrigger/issues/44 seems to suggest that 

The good news here is that I believe your issues are fixed in version 3, which integrates pgtrigger with migrations. I.e. there is no more reason to prune or do this stuff to detect changes

I would still like to get to the bottom of this though, so any more info on your setup would be nice.

One final question. Does `manage.py pgtrigger ls` also show a bunch of pruned triggers for the other schemas? Would you assume that `ls` and other commands would only operate on the current schema?

**Comment by wesleykendall** [comment]
@ooprala this bug is fixed in https://github.com/Opus10/django-pgtrigger/pull/80

Along with re-architecting multi-db support, I added some test cases of DBs that set the search path. I was able to reproduce your bug and solved it by ignoring triggers that were not in the search path.

When released, there will be a new part of the docs that discusses multi-schema setup. If you define the ``search_path`` in the ``settings.DATABASES``, you can just use the multi-db functionality. If you use something like `django-tenants`, triggers should still be installed fine relative to the search path (just like tables), and you can also supply schemas to the management commands to dynamically alter the search path.

For example, `manage.py pgtrigger ls -s my_schema -s public` will list trigger installation status relative to the ``my_schema,public`` search path

**Comment by wesleykendall** [comment]
Fixed in version 4
